File: services/customer-service/src/main/java/com/banking/platform/customerservice/domain/model/Customer.java
package com.banking.platform.customerservice.domain.model;

import lombok.*;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Customer {
    private UUID id;
    private String firstName;
    private String lastName;
    private String email;
    private String phoneNumber;
    private String taxId;
    private LocalDate dateOfBirth;
    private Address address;
    private CustomerStatus status;
    private Instant createdAt;
    private Instant updatedAt;

    public enum CustomerStatus {
        PENDING_VERIFICATION, ACTIVE, SUSPENDED, BLACKLISTED, CLOSED
    }

    public String getFullName() {
        return firstName + " " + lastName;
    }

    public boolean canTransact() {
        return this.status == CustomerStatus.ACTIVE;
    }
}

File: frontend/src/types/customer.ts
import { CustomerStatus, AccountType, AccountStatus } from './enums';

export interface Customer {
  id: string;
  firstName: string;
  lastName: string;
  fullName: string;
  email: string;
  phoneNumber?: string;
  taxId: string;
  dateOfBirth?: string;
  status: CustomerStatus;
  address?: Address;
  createdAt: string;
  updatedAt: string;
}

export interface Address {
  id: string;
  street: string;
  city: string;
  state: string;
  zipCode: string;
  country: string;
}

export interface CreateCustomerRequest {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber?: string;
  taxId: string;
  dateOfBirth?: string;
  address?: {
    street: string;
    city: string;
    state: string;
    zipCode: string;
    country: string;
  };
}

export interface UpdateCustomerRequest {
  firstName?: string;
  lastName?: string;
  email?: string;
  phoneNumber?: string;
  address?: Address;
}
